src/hotspot/share/opto/vectorIntrinsics.cpp

```diff
@@ -1513,18 +1553,44 @@ bool LibraryCallKit::inline_vector_broadcast_int() {
     }
     return false; // operation not supported
   }
+  Node* cnt  = argument(5);
   ciKlass* vbox_klass = vector_klass->const_oop()->as_instance()->java_lang_Class_klass();
   const TypeInstPtr* vbox_type = TypeInstPtr::make_exact(TypePtr::NotNull, vbox_klass);
-
-  if (!arch_supports_vector(sopc, num_elem, elem_bt, VecMaskNotUsed, true /*has_scalar_args*/)) {
-    if (C->print_intrinsics()) {
-      tty->print_cr("  ** not supported: arity=0 op=int/%d vlen=%d etype=%s ismask=no",
-                    sopc, num_elem, type2name(elem_bt));
+  const TypeInt* cnt_type = cnt->bottom_type()->isa_int();
+  bool is_const_rotate = is_rotate && cnt_type && cnt_type->is_con() &&
+                         -0x80 <= cnt_type->get_con() && cnt_type->get_con() < 0x80;
+  if (is_rotate) {
+    if (!arch_supports_vector_rotate(sopc, num_elem, elem_bt, !is_const_rotate)) {
```

Comment:
What is the relationship between check_bcast and !is_const_rotate? Some comments here on this would help.

Reply:
Constant shift value is an optimizing case since AVX512 offers instructions which directly accept constant shifts in the range (-256, 255). Similar handling is done in SLP.
https://github.com/openjdk/jdk/blob/master/src/hotspot/share/opto/superword.cpp#L2493

But I feel this is very X86 specific check in generic code, so moving decision to a new target specific matcher routine.